src/java/picard/sam/RevertSam.java

```diff
@@ -240,37 +250,96 @@ protected int doWork() {
         if (!sanitizing) {
             out.close();
         } else {
+            final Map<SAMReadGroupRecord, FastqQualityFormat> readGroupToFormat;
+            try {
+                readGroupToFormat = createReadGroupFormatMap(inHeader, REFERENCE_SEQUENCE, VALIDATION_STRINGENCY, INPUT, RESTORE_ORIGINAL_QUALITIES);
+            } catch (PicardException e) {
+                log.error(e.getMessage());
+                return -1;
+            }
 
-            long total = 0, discarded = 0;
-            final PeekableIterator<SAMRecord> iterator = new PeekableIterator<SAMRecord>(sorter.iterator());
-            final Map<SAMReadGroupRecord, FastqQualityFormat> readGroupToFormat = new HashMap<SAMReadGroupRecord, FastqQualityFormat>();
+            long[] sanitizeResults = sanitize(readGroupToFormat, sorter, out);
```

Comment:
I don't know if this is the case for @dshiga but I learned recently that for people using the dark theme it's kinda hard to spot those warnings, the yellow shade they use there is subtle.